File: backend/src/main/java/slib/com/example/entity/users/UserSetting.java
package slib.com.example.entity.users;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "user_settings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserSetting {

    @Id
    @Column(name = "user_id")
    private UUID userId;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "user_id")
    @JsonIgnore
    @ToString.Exclude
    private User user;

    @Column(name = "is_hce_enabled")
    private Boolean isHceEnabled;

    @Column(name = "is_ai_recommend_enabled")
    private Boolean isAiRecommendEnabled;

    @Column(name = "is_booking_remind_enabled")
    private Boolean isBookingRemindEnabled;

    @Column(name = "theme_mode", length = 20)
    private String themeMode;

    @Column(name = "language_code", length = 10)
    private String languageCode;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: mobile/lib/models/user_setting.dart
class UserSetting {
  final String userId;
  final bool isHceEnabled;
  final bool isAiRecommendEnabled;
  final bool isBookingRemindEnabled;
  final String themeMode;
  final String languageCode;

  UserSetting({
    required this.userId,
    required this.isHceEnabled,
    required this.isAiRecommendEnabled,
    required this.isBookingRemindEnabled,
    required this.themeMode,
    required this.languageCode,
  });

  factory UserSetting.fromJson(Map<String, dynamic> json) {
    return UserSetting(
      userId: json['userId']?.toString() ?? '',
      isHceEnabled: json['isHceEnabled'] ?? false,
      isAiRecommendEnabled: json['isAiRecommendEnabled'] ?? true,
      isBookingRemindEnabled: json['isBookingRemindEnabled'] ?? true,
      themeMode: json['themeMode'] ?? 'light',
      languageCode: json['languageCode'] ?? 'vi',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'isHceEnabled': isHceEnabled,
      'isAiRecommendEnabled': isAiRecommendEnabled,
      'isBookingRemindEnabled': isBookingRemindEnabled,
      'themeMode': themeMode,
      'languageCode': languageCode,
    };
  }

  UserSetting copyWith({
    bool? isHceEnabled,
    bool? isAiRecommendEnabled,
    bool? isBookingRemindEnabled,
    String? themeMode,
    String? languageCode,
  }) {
    return UserSetting(
      userId: userId,
      isHceEnabled: isHceEnabled ?? this.isHceEnabled,
      isAiRecommendEnabled: isAiRecommendEnabled ?? this.isAiRecommendEnabled,
      isBookingRemindEnabled:
          isBookingRemindEnabled ?? this.isBookingRemindEnabled,
      themeMode: themeMode ?? this.themeMode,
      languageCode: languageCode ?? this.languageCode,
    );
  }
}
